### PR TITLE
feat:  add i18n support

### PR DESCRIPTION
**Pull request by Quilljou**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# http://editorconfig.org
+root = true
```
Comment (futantan) [inline .editorconfig:2]: there is a PR https://github.com/futantan/OpenGpt/pull/10 which is about the format and eslint, please remove this file and we will use that instead, or add this to your .gitignore file

**Review by futantan — CHANGES_REQUESTED** [review]

**Comment by futantan** [comment]
awesome! thank you @Quilljou !

File: .editorconfig (hunk 1 above)
Comment (Quilljou) [reply .editorconfig]: done

**Comment by Quilljou** [comment]
btw, the tool https://chatgpt-i18n.vercel.app/translate is awesome

**Comment by futantan** [comment]
thanks for the sharing!
I have two concerns about this and like to know your ideas:
- this project is in very early stage, and most of the users speak Chinese
- We will absolutely add i18n soon, but I'm worried that this will burden the work for the developer (mostly me)

I really love this PR and want to move this forward, how about this:
- hard code language to zh-cn for now and merge the PR
- another PR in the future to enable the language detection.

**Comment by futantan** [comment]
This PR will close https://github.com/futantan/OpenGpt/issues/6

**Comment by Quilljou** [comment]
> thanks for the sharing! I have two concerns about this and like to know your ideas:
> 
> * this project is in very early stage, and most of the users speak Chinese
> * We will absolutely add i18n soon, but I'm worried that this will burden the work for the developer (mostly me)
> 
> I really love this PR and want to move this forward, how about this:
> 
> * hard code language to zh-cn for now and merge the PR
> * another PR in the future to enable the language detection.

Does hard code mean that the text in future new features will be hardcoded in Chinese instead of being written under locales?

**Comment by futantan** [comment]
no.  I mean we will write it under locales of course, but only use the Chinese locale for now

**Comment by Quilljou** [comment]
> no. I mean we will write it under locales of course, but only use the Chinese locale for now
Great. I'm currently developing a tool to simplify the burden of i18n work.

**Comment by Quilljou** [comment]
@futantan All conflicts are resolved

**Comment by futantan** [comment]
hi @Quilljou I created another PR to commit based on your history, which contains:
- support zh-TW
- add discord link

Please correct me it this is not the right way to update based on your code, thank you. 
https://github.com/futantan/OpenGpt/pull/33

**Comment by Quilljou** [comment]
Screenshot 2023-03-12 at 19 54 30
i18n seems broken online, but it works in my local dev environment

**Comment by futantan** [comment]
can you help to refresh the page and check it again? 
I think it's fixed

**Comment by Quilljou** [comment]
I open it in my phone, it's still broken

**Comment by futantan** [comment]
oh, it’s broken… could you help to fix it?
I’m outside from my computer

**Comment by Quilljou** [comment]
OK, let me check

**Comment by Quilljou** [comment]
Maybe something is wrong with Vecel, it works fine in my local dev environment with the latest code from upstream main branch

**Comment by futantan** [comment]
same here.
but it's still broken when I redeployed

**Comment by futantan** [comment]
I disabled other language...
https://github.com/futantan/OpenGpt/pull/34
let's enable them when we find the root cause

**Comment by futantan** [comment]
This did not fix the issue...
I rollback the deployment...

do you have any ideas about this? I haven't used these libs before

**Comment by futantan** [comment]
sorry, I reverted the i18n related code, let's find out the root cause first

**Comment by Quilljou** [comment]
Screenshot 2023-03-12 at 21 00 15
try to remove all default locale, and 
Screenshot 2023-03-12 at 21 01 34
react-i18n -> next-i18next, deploy on vercel preview enviroment to check

**Comment by futantan** [comment]
as your suggestion, please check the latest 3 commits on main brach...
still not working

**Comment by futantan** [comment]
do you mind add me as your wechat friend? might a better way to communicate in this case

**Comment by futantan** [comment]
sorry I can't figure it out, and reverted all the related code for now.
Let's bring this back in the future once we find the root case.

Thanks for your contribution, my bad... @Quilljou

**Comment by Quilljou** [comment]
I apologize for the delayed response. I will review it later and test it on my Vercel account.
I have sent my WeChat ID to your Gmail. @futantan

**Pull request merged**